apps/mymedicare_cb/authorization.py

```diff
@@ -59,127 +82,224 @@ def basic_auth(self):
+            if self.signout_status_code != status.HTTP_302_FOUND:
+                log_authenticate_start(auth_flow_dict, "FAIL",
+                                       "SLSx signout response_code = {code}."
+                                       " Expecting HTTP_302_FOUND.".format(code=self.signout_status_code),
+                                       slsx_client=self)
+                raise BBMyMedicareSLSxSignoutException(settings.MEDICARE_ERROR_MSG)
             response.raise_for_status()
         except (requests.exceptions.HTTPError, requests.exceptions.ConnectionError) as e:
-            log_authenticate_start(auth_flow_dict, "FAIL",
-                                   "SLSx signout error {reason}".format(reason=e))
-            raise BBMyMedicareSLSxSignoutException(settings.MEDICARE_ERROR_MSG)
+            if self.signout_status_code != status.HTTP_302_FOUND:
+                self.signout_status_mesg = e
+                log_authenticate_start(auth_flow_dict, "FAIL",
+                                       "SLSx signout error {reason}".format(reason=e),
+                                       slsx_client=self)
+                raise BBMyMedicareSLSxSignoutException(settings.MEDICARE_ERROR_MSG)
+
+    def validate_user_signout(self, request):
+        """
+        Performs a call to self.get_user_info to validate that the bene is
+        signed out. When NOT signed out, an exception is thrown.
+        This assumes the bene is signed out, if the userinfo endpoint returns
+        a HTTP_403_FORBIDDEN respose, since the auth_token is no longer valid.
+        """
+        headers = self.slsx_common_headers(request)
+        headers.update(self.auth_header())
+
+        auth_flow_dict = get_session_auth_flow_trace(request)
+
+        try:
+            response = requests.get(self.userinfo_endpoint + "/" + self.user_id,
+                                    headers=headers,
+                                    allow_redirects=False,
+                                    verify=self.verify_ssl,
+                                    hooks={
+                                        'response': [
+                                            response_hook_wrapper(sender=SLSxUserInfoResponse,
+                                                                  auth_flow_dict=auth_flow_dict)]})
+            self.validate_signout_status_code = response.status_code
+            if self.validate_signout_status_code != status.HTTP_403_FORBIDDEN:
+                log_authenticate_start(auth_flow_dict, "FAIL",
+                                       "SLSx validate signout response_code = {code}."
+                                       " Expecting HTTP_403_FORBIDDEN.".format(code=self.validate_signout_status_code),
+                                       slsx_client=self)
+                raise BBMyMedicareSLSxValidateSignoutException(settings.MEDICARE_ERROR_MSG)
+            response.raise_for_status()
```

Comment:
I'm a little curious about how this works, as we will raise an exception for anything that isn't an HTTP 403, but then reponse.raise_for_status() will likely raise an exception over the expected HTTP 403.  Am I missing something here?